doc/openapi-user-guide.md

```diff
@@ -92,6 +93,82 @@ spec:
 
 [OpenAPI CRD Reference](openapi-reference.md) for more info.
 
+### OpenID Connect and OAuth2 example
+
+3scale requires some additional information that are not included 
+in the OpenAPI spec, therefore those data need to be provided in the OpenAPI CR. Specifically:
+- OpenID Connect Issuer Type which will default to rest but can be overridden from the OpenAPI CR
+- OpenID Connect Issuer. 3scale requires that the issuer URL must include a client secret.
+- Flows object. When the sec scheme is oauth2, the flows are provided by the OpenAPI doc. However, for openIdConnect security scheme, the OpenAPI doc does not provide the flows. In that case, the OpenAPI CR can provide those.
+
+#### OIDC Issuer Secret
+Secret name:  **oidc-issuer-client-secret**  
+Example of the secret:
+```shell
+cat oidc-issuer-client-secret.yaml 
+```
+```yaml
+kind: Secret
+apiVersion: v1
+metadata: 
+  name: oidc-issuer-client-secret
+  namespace: 3scale-test
+data:
+  secret: XXXXXXXXXXXXXXXXXXXXX
+type: Opaque
+```
+To create secret:
+```shell
+oc apply -f oidc-issuer-client-secret.yaml 
+```
+
+**NOTEs**
+- **oidc-issuer-client-secret** `name is Not configurable` Please use this secret name as noted.
+- The secret value will be taken from the Issuer Client secret. If you are using RHSSO - the secret could be found at `Realm/Clients/ClientID/Credentials/Secret`.
+
+#### OpenAPI CR example for OIDC and oauth2
+```yaml
+apiVersion: capabilities.3scale.net/v1beta1
+kind: OpenAPI
+metadata:
+  generation: 1
+  name: openapi-example
+spec:
+  openapiRef:
+    url: https://raw.githubusercontent.com/valerymo/OpenAPI-Specification/test/examples/v3.0/petstore.yaml
+  oidc:
+    issuerType: keycloak
+    issuerEndpoint: https://3scale-zync:some-secret@keycloak-rhsso-test.apps.xxxxx.xxxx.s1.devshift.org/auth/realms/petstore
```

Comment:
Have you considered the following approach?  
```yaml
spec:
  oidc:
    issuerEndpointRef:
      name: my-secret
```

Then a secret
```yaml
kind: Secret
apiVersion: v1
metadata: 
  name: my-secret
  namespace: 3scale-test
data:
  issuerEndpoint: https://3scale-zync:some-secret@keycloak-rhsso-test.apps.xxxxx.xxxx.s1.devshift.org/auth/realms/petstore
type: Opaque
```

Reply:
> Have you considered the following approach?
> 
> ```yaml
> spec:
>   oidc:
>     issuerEndpointRef:
>       name: my-secret
> ```
> 
> Then a secret
> 
> ```yaml
> kind: Secret
> apiVersion: v1
> metadata: 
>   name: my-secret
>   namespace: 3scale-test
> data:
>   issuerEndpoint: https://3scale-zync:some-secret@keycloak-rhsso-test.apps.xxxxx.xxxx.s1.devshift.org/auth/realms/petstore
> type: Opaque
> ```

done. Thank you